app/search/page.tsx

```
@@ -1,7 +1,192 @@
+  useEffect(() => {
+    if (!hasMore || loading) return;
+    const observer = new IntersectionObserver((entries) => {
+      if (entries[0].isIntersecting) {
+        setLoading(true);
+        let filtered = allHadiths;
+        if (source) filtered = filtered.filter((h) => h.source === source);
+        if (chapter) filtered = filtered.filter((h) => h.chapter === chapter);
+        if (debouncedNarrator)
+          filtered = filtered.filter(
+            (h) =>
+              h.narrator_name && h.narrator_name.includes(debouncedNarrator),
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)